Confusing error from "flutter ios build" when Xcode account session expires

```
dalmingmunming-ui-MacBook-Pro:hugin jyd$ flutter build ios
Building com.swfact.hugin for device (ios-release)...
Found saved certificate choice "iPhone Developer: Yaedam Jung (YZFXKH3K7N)". To clear, use "flutter config".
Signing iOS app for device d                                                                                                            Xcode build done.                                           18.4s                                                                       Failed to build iOS app                                                                                             
Error output from Xcode build:
↳
    2019-05-15 20:40:53.399 xcodebuild[51263:426457]  DVTPortal: Service '<DVTPortalViewDeveloperService: 0x7fee31678fc0;
    action='viewDeveloper'>' encountered an unexpected result code from the portal ('1100')
    2019-05-15 20:40:53.399 xcodebuild[51263:426457]  DVTPortal: Error:
    Error Domain=DVTPortalServiceErrorDomain Code=1100 "Your session has expired.  Please log in." UserInfo={payload=<CFBasicHash
    0x7fee319b1790 [0x7fff8686a8f0]>{type = mutable dict, count = 9,
    entries =>
        0 : responseId = <CFString 0x7fee32a25420 [0x7fff8686a8f0]>{contents = "3cb59e42-3744-41f1-a847-979848f00354"}
        2 : <CFString 0x7fff868c7598 [0x7fff8686a8f0]>{contents = "protocolVersion"} = QH65B2
        3 : <CFString 0x7fee32a25300 [0x7fff8686a8f0]>{contents = "requestUrl"} = <CFString 0x7fee32a25340 [0x7fff8686a8f0]>{contents =
        "https://developerservices2.apple.com/services/QH65B2/viewDeveloper.action"}
        6 : <CFString 0x7fee32a25220 [0x7fff8686a8f0]>{contents = "userLocale"} = en_US
        8 : resultCode = <CFNumber 0x56c360f27e5665f3 [0x7fff8686a8f0]>{value = +1100, type = kCFNumberSInt64Type}
        9 : userString = <CFString 0x7fee32a24fc0 [0x7fff8686a8f0]>{contents = "Your session has expired.  Please log in."}
        10 : <CFString 0x7fee323e57b0 [0x7fff8686a8f0]>{contents = "resultString"} = <CFString 0x7fee32a25060 [0x7fff8686a8f0]>{contents =
        "authentication.failed"}
        11 : httpCode = <CFNumber 0x56c360f27e52e1f3 [0x7fff8686a8f0]>{value = +200, type = kCFNumberSInt64Type}
        12 : <CFString 0x7fee32a250f0 [0x7fff8686a8f0]>{contents = "creationTimestamp"} = <CFString 0x7fee32a250c0
        [0x7fff8686a8f0]>{contents = "2019-05-15T11:40:53Z"}
    }
    , NSLocalizedDescription=Your session has expired.  Please log in.}
    2019-05-15 20:40:54.170 xcodebuild[51263:426457]  DVTPortal: Service '<DVTPortalViewDeveloperService: 0x7fee31012990;
    action='viewDeveloper'>' encountered an unexpected result code from the portal ('1100')
    2019-05-15 20:40:54.170 xcodebuild[51263:426457]  DVTPortal: Error:
    Error Domain=DVTPortalServiceErrorDomain Code=1100 "Your session has expired.  Please log in." UserInfo={payload=<CFBasicHash
    0x7fee335d4e30 [0x7fff8686a8f0]>{type = mutable dict, count = 9,
    entries =>
        0 : responseId = <CFString 0x7fee338f9bf0 [0x7fff8686a8f0]>{contents = "683a8b4d-2624-4659-84f1-7f2a7f99b2b2"}
        2 : <CFString 0x7fff868c7598 [0x7fff8686a8f0]>{contents = "protocolVersion"} = QH65B2
        3 : <CFString 0x7fee338f97e0 [0x7fff8686a8f0]>{contents = "requestUrl"} = <CFString 0x7fee33c0b4a0 [0x7fff8686a8f0]>{contents =
        "https://developerservices2.apple.com/services/QH65B2/viewDeveloper.action"}
        6 : <CFString 0x7fee338f6610 [0x7fff8686a8f0]>{contents = "userLocale"} = en_US
        8 : resultCode = <CFNumber 0x56c360f27e5665f3 [0x7fff8686a8f0]>{value = +1100, type = kCFNumberSInt64Type}
        9 : userString = <CFString 0x7fee335dba70 [0x7fff8686a8f0]>{contents = "Your session has expired.  Please log in."}
        10 : <CFString 0x7fee335dca80 [0x7fff8686a8f0]>{contents = "resultString"} = <CFString 0x7fee335dd440 [0x7fff8686a8f0]>{contents =
        "authentication.failed"}
        11 : httpCode = <CFNumber 0x56c360f27e52e1f3 [0x7fff8686a8f0]>{value = +200, type = kCFNumberSInt64Type}
        12 : <CFString 0x7fee33553870 [0x7fff8686a8f0]>{contents = "creationTimestamp"} = <CFString 0x7fee335e42e0
        [0x7fff8686a8f0]>{contents = "2019-05-15T11:40:54Z"}
    }
    , NSLocalizedDescription=Your session has expired.  Please log in.}
    2019-05-15 20:40:54.730 xcodebuild[51263:426457]  DVTPortal: Service '<DVTPortalViewDeveloperService: 0x7fee34216390;
    action='viewDeveloper'>' encountered an unexpected result code from the portal ('1100')
    2019-05-15 20:40:54.731 xcodebuild[51263:426457]  DVTPortal: Error:
    Error Domain=DVTPortalServiceErrorDomain Code=1100 "Your session has expired.  Please log in." UserInfo={payload=<CFBasicHash
    0x7fee34147ff0 [0x7fff8686a8f0]>{type = mutable dict, count = 9,
    entries =>
        0 : responseId = <CFString 0x7fee3417d720 [0x7fff8686a8f0]>{contents = "4b33ba2d-94a3-4b0b-b4aa-0d8bc8e4a899"}
        2 : <CFString 0x7fff868c7598 [0x7fff8686a8f0]>{contents = "protocolVersion"} = QH65B2
        3 : <CFString 0x7fee341952f0 [0x7fff8686a8f0]>{contents = "requestUrl"} = <CFString 0x7fee34152f10 [0x7fff8686a8f0]>{contents =
        "https://developerservices2.apple.com/services/QH65B2/viewDeveloper.action"}
        6 : <CFString 0x7fee333af890 [0x7fff8686a8f0]>{contents = "userLocale"} = en_US
        8 : resultCode = <CFNumber 0x56c360f27e5665f3 [0x7fff8686a8f0]>{value = +1100, type = kCFNumberSInt64Type}
        9 : userString = <CFString 0x7fee3414efc0 [0x7fff8686a8f0]>{contents = "Your session has expired.  Please log in."}
        10 : <CFString 0x7fee33b33aa0 [0x7fff8686a8f0]>{contents = "resultString"} = <CFString 0x7fee337ba150 [0x7fff8686a8f0]>{contents =
        "authentication.failed"}
        11 : httpCode = <CFNumber 0x56c360f27e52e1f3 [0x7fff8686a8f0]>{value = +200, type = kCFNumberSInt64Type}
        12 : <CFString 0x7fee333af830 [0x7fff8686a8f0]>{contents = "creationTimestamp"} = <CFString 0x7fee333af800
        [0x7fff8686a8f0]>{contents = "2019-05-15T11:40:54Z"}
    }
    , NSLocalizedDescription=Your session has expired.  Please log in.}
    ** BUILD FAILED **
```

Xcode's output:
↳
```
    === BUILD TARGET Runner OF PROJECT Runner WITH CONFIGURATION Release ===
    ld: warning: ignoring file /Users/jyd/Documents/project/hugin/ios/Flutter/App.framework/App, file was built for x86_64 which is not
    the architecture being linked (arm64): /Users/jyd/Documents/project/hugin/ios/Flutter/App.framework/App
    === BUILD TARGET Runner OF PROJECT Runner WITH CONFIGURATION Release ===
    /* com.apple.actool.document.warnings */
    /Users/jyd/Documents/project/hugin/ios/Runner/Assets.xcassets:./AppIcon.appiconset/(null)[2d][sd.png]: warning: The app icon set
    "AppIcon" has an unassigned child.
    /Users/jyd/Documents/project/hugin/ios/Runner/Assets.xcassets:./AppIcon.appiconset/(null)[2d][sd.png]: warning: The app icon set
    "AppIcon" has an unassigned child.
    /* com.apple.actool.document.notices */
    /Users/jyd/Documents/project/hugin/ios/Runner/Assets.xcassets:./AppIcon.appiconset/[][ipad][76x76][][][1x][][]: notice: 76x76@1x app
    icons only apply to iPad apps targeting releases of iOS prior to 10.0.
    /* com.apple.actool.compilation-results */
    /Users/jyd/Documents/project/hugin/build/ios/Release-iphoneos/Runner.app/AppIcon60x60@2x.png
    /Users/jyd/Documents/project/hugin/build/ios/Release-iphoneos/Runner.app/AppIcon76x76@2x~ipad.png
    /Users/jyd/Documents/project/hugin/build/ios/Release-iphoneos/Runner.app/Assets.car
    /Users/jyd/Library/Developer/Xcode/DerivedData/Runner-agglnejewxhbrcfwdyrekfylaeth/Build/Intermediates.noindex/Runner.build/Release-
    iphoneos/Runner.build/assetcatalog_generated_info.plist

    === BUILD TARGET Runner OF PROJECT Runner WITH CONFIGURATION Release ===
    Non-fat binary /Users/jyd/Documents/project/hugin/build/ios/Release-iphoneos/Runner.app/Frameworks/App.framework/App is not arm64.
    Running lipo -info:
    Non-fat file: /Users/jyd/Documents/project/hugin/build/ios/Release-iphoneos/Runner.app/Frameworks/App.framework/App is architecture:
    x86_64
    Command /bin/sh failed with exit code 1

Encountered error while building for device.
```

--- a/dev/devicelab/bin/tasks/module_test_ios.dart
+++ b/dev/devicelab/bin/tasks/module_test_ios.dart
@@ -548,8 +548,7 @@ end
       );
 
       if (!xcodebuildOutput.contains('flutter --verbose --local-engine-src-path=bogus assemble') || // Verbose output
-          !xcodebuildOutput.contains('Unable to detect a Flutter engine build directory in bogus') ||
-          !xcodebuildOutput.contains('Command PhaseScriptExecution failed with a nonzero exit code')) {
+          !xcodebuildOutput.contains('Unable to detect a Flutter engine build directory in bogus')) {
         return TaskResult.failure('Host Objective-C app build succeeded though flutter script failed');
       }
 

--- a/packages/flutter_tools/bin/xcode_backend.dart
+++ b/packages/flutter_tools/bin/xcode_backend.dart
@@ -97,11 +97,17 @@ class Context {
     if (verbose) {
       print((result.stdout as String).trim());
     }
-    if ((result.stderr as String).isNotEmpty) {
-      echoError((result.stderr as String).trim());
+    final String resultStderr = result.stderr.toString().trim();
+    if (resultStderr.isNotEmpty) {
+      final StringBuffer errorOutput = StringBuffer();
+      if (result.exitCode != 0) {
+        // "error:" prefix makes this show up as an Xcode compilation error.
+        errorOutput.write('error: ');
+      }
+      errorOutput.write(resultStderr);
+      echoError(errorOutput.toString());
     }
     if (!allowFail && result.exitCode != 0) {
-      stderr.write('${result.stderr}\n');
       throw Exception(
         'Command "$bin ${args.join(' ')}" exited with code ${result.exitCode}',
       );

--- a/packages/flutter_tools/lib/src/build_system/build_system.dart
+++ b/packages/flutter_tools/lib/src/build_system/build_system.dart
@@ -871,13 +871,11 @@ class _BuildInstance {
         ErrorHandlingFileSystem.deleteIfExists(previousFile);
       }
     } on Exception catch (exception, stackTrace) {
-      // TODO(zanderso): throw specific exception for expected errors to mark
-      // as non-fatal. All others should be fatal.
       node.target.clearStamp(environment);
       succeeded = false;
       skipped = false;
       exceptionMeasurements[node.target.name] = ExceptionMeasurement(
-          node.target.name, exception, stackTrace);
+          node.target.name, exception, stackTrace, fatal: true);
     } finally {
       resource.release();
       stopwatch.stop();

--- a/packages/flutter_tools/lib/src/commands/assemble.dart
+++ b/packages/flutter_tools/lib/src/commands/assemble.dart
@@ -347,7 +347,7 @@ class AssembleCommand extends FlutterCommand {
       for (final ExceptionMeasurement measurement in result.exceptions.values) {
         if (measurement.fatal || globals.logger.isVerbose) {
           globals.printError('Target ${measurement.target} failed: ${measurement.exception}',
-            stackTrace: measurement.stackTrace
+            stackTrace: globals.logger.isVerbose ? measurement.stackTrace : null,
           );
         }
       }

--- a/packages/flutter_tools/lib/src/ios/mac.dart
+++ b/packages/flutter_tools/lib/src/ios/mac.dart
@@ -414,14 +414,6 @@ Future<XcodeBuildResult> buildXcodeProject({
   }
   if (buildResult != null && buildResult.exitCode != 0) {
     globals.printStatus('Failed to build iOS app');
-    if (buildResult.stderr.isNotEmpty) {
-      globals.printStatus('Error output from Xcode build:\n↳');
-      globals.printStatus(buildResult.stderr, indent: 4);
-    }
-    if (buildResult.stdout.isNotEmpty) {
-      globals.printStatus("Xcode's output:\n↳");
-      globals.printStatus(buildResult.stdout, indent: 4);
-    }
     return XcodeBuildResult(
       success: false,
       stdout: buildResult.stdout,
@@ -738,7 +730,7 @@ bool _handleIssues(XCResult? xcResult, Logger logger, XcodeBuildExecution? xcode
 
   if (requiresProvisioningProfile) {
     logger.printError(noProvisioningProfileInstruction, emphasis: true);
-  } else if (_missingDevelopmentTeam(xcodeBuildExecution)) {
+  } else if ((!issueDetected || hasProvisioningProfileIssue) && _missingDevelopmentTeam(xcodeBuildExecution)) {
     issueDetected = true;
     logger.printError(noDevelopmentTeamInstruction, emphasis: true);
   } else if (hasProvisioningProfileIssue) {
@@ -782,11 +774,21 @@ bool _needUpdateSigningIdentifier(XcodeBuildExecution? xcodeBuildExecution) {
 //
 // As detecting issues in stdout is not usually accurate, this should be used as a fallback when other issue detecting methods failed.
 void _parseIssueInStdout(XcodeBuildExecution xcodeBuildExecution, Logger logger, XcodeBuildResult result) {
+  final String? stderr = result.stderr;
+  if (stderr != null && stderr.isNotEmpty) {
+    logger.printStatus('Error output from Xcode build:\n↳');
+    logger.printStatus(stderr, indent: 4);
+  }
+  final String? stdout = result.stdout;
+  if (stdout != null && stdout.isNotEmpty) {
+    logger.printStatus("Xcode's output:\n↳");
+    logger.printStatus(stdout, indent: 4);
+  }
+
   if (xcodeBuildExecution.environmentType == EnvironmentType.physical
       // May need updating if Xcode changes its outputs.
       && (result.stdout?.contains('requires a provisioning profile. Select a provisioning profile in the Signing & Capabilities editor') ?? false)) {
     logger.printError(noProvisioningProfileInstruction, emphasis: true);
-    return;
   }
 }
 

--- a/packages/flutter_tools/lib/src/ios/xcresult.dart
+++ b/packages/flutter_tools/lib/src/ios/xcresult.dart
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'package:meta/meta.dart';
+
 import '../../src/base/process.dart';
 import '../../src/convert.dart' show json;
 import '../../src/macos/xcode.dart';
@@ -112,6 +114,20 @@ class XCResult {
     );
   }
 
+  /// Create a [XCResult] with constructed [XCResultIssue]s for testing.
+  @visibleForTesting
+  factory XCResult.test({
+    List<XCResultIssue>? issues,
+    bool? parseSuccess,
+    String? parsingErrorMessage,
+  }) {
+    return XCResult._(
+      issues: issues ?? const <XCResultIssue>[],
+      parseSuccess: parseSuccess ?? true,
+      parsingErrorMessage: parsingErrorMessage,
+    );
+  }
+
   XCResult._({
     this.issues = const <XCResultIssue>[],
     this.parseSuccess = true,
@@ -180,6 +196,24 @@ class XCResultIssue {
       }
     }
 
+    return XCResultIssue._(
+      type: type,
+      subType: subType,
+      message: message,
+      location: location,
+      warnings: warnings,
+    );
+  }
+
+  /// Create a [XCResultIssue] without JSON parsing for testing.
+  @visibleForTesting
+  factory XCResultIssue.test({
+    XCResultIssueType type = XCResultIssueType.error,
+    String? subType,
+    String? message,
+    String? location,
+    List<String> warnings = const <String>[],
+  }) {
     return XCResultIssue._(
       type: type,
       subType: subType,

--- a/packages/flutter_tools/test/commands.shard/hermetic/assemble_test.dart
+++ b/packages/flutter_tools/test/commands.shard/hermetic/assemble_test.dart
@@ -135,13 +135,13 @@ void main() {
   testUsingContext('flutter assemble does not log stack traces during build failure', () async {
     final CommandRunner<void> commandRunner = createTestCommandRunner(AssembleCommand(
       buildSystem: TestBuildSystem.all(BuildResult(success: false, exceptions: <String, ExceptionMeasurement>{
-        'hello': ExceptionMeasurement('hello', 'bar', stackTrace),
+        'hello': ExceptionMeasurement('hello', 'bar', stackTrace, fatal: true),
       }))
     ));
 
     await expectLater(commandRunner.run(<String>['assemble', '-o Output', 'debug_macos_bundle_flutter_assets']),
       throwsToolExit());
-    expect(testLogger.errorText, isNot(contains('bar')));
+    expect(testLogger.errorText, contains('Target hello failed: bar'));
     expect(testLogger.errorText, isNot(contains(stackTrace.toString())));
   }, overrides: <Type, Generator>{
     Cache: () => Cache.test(processManager: FakeProcessManager.any()),

--- a/packages/flutter_tools/test/commands.shard/hermetic/build_ios_test.dart
+++ b/packages/flutter_tools/test/commands.shard/hermetic/build_ios_test.dart
@@ -372,7 +372,7 @@ void main() {
       XcodeProjectInterpreter: () => FakeXcodeProjectInterpreterWithBuildSettings(),
     });
 
-    testUsingContext('Display xcresult issues on console if parsed.', () async {
+    testUsingContext('Display xcresult issues on console if parsed, suppress Xcode output', () async {
       final BuildCommand command = BuildCommand();
 
       createMinimalMockProjectFiles();
@@ -384,13 +384,16 @@ void main() {
 
       expect(testLogger.errorText, contains("Use of undeclared identifier 'asdas'"));
       expect(testLogger.errorText, contains('/Users/m/Projects/test_create/ios/Runner/AppDelegate.m:7:56'));
+      expect(testLogger.statusText, isNot(contains("Xcode's output")));
+      expect(testLogger.statusText, isNot(contains('Lots of spew from Xcode')));
     }, overrides: <Type, Generator>{
       FileSystem: () => fileSystem,
       ProcessManager: () => FakeProcessManager.list(<FakeCommand>[
         xattrCommand,
         setUpFakeXcodeBuildHandler(exitCode: 1, onRun: () {
           fileSystem.systemTempDirectory.childDirectory(_xcBundleFilePath).createSync();
-        }),
+        }, stdout: 'Lots of spew from Xcode',
+        ),
         setUpXCResultCommand(stdout: kSampleResultJsonWithIssues),
         setUpRsyncCommand(),
       ]),
@@ -620,7 +623,6 @@ Runner requires a provisioning profile. Select a provisioning profile in the Sig
       Platform: () => macosPlatform,
       XcodeProjectInterpreter: () => FakeXcodeProjectInterpreterWithBuildSettings(developmentTeam: null),
     });
-
 
     testUsingContext('xcresult did not detect issue but detected by stdout.', () async {
       final BuildCommand command = BuildCommand();

--- a/packages/flutter_tools/test/general.shard/ios/mac_test.dart
+++ b/packages/flutter_tools/test/general.shard/ios/mac_test.dart
@@ -13,6 +13,7 @@ import 'package:flutter_tools/src/cache.dart';
 import 'package:flutter_tools/src/ios/code_signing.dart';
 import 'package:flutter_tools/src/ios/iproxy.dart';
 import 'package:flutter_tools/src/ios/mac.dart';
+import 'package:flutter_tools/src/ios/xcresult.dart';
 import 'package:flutter_tools/src/project.dart';
 import 'package:flutter_tools/src/reporting/reporting.dart';
 import 'package:test/fake.dart';
@@ -265,49 +266,8 @@ Xcode's output:
 ↳
     blah
 
-    === CLEAN TARGET url_launcher OF PROJECT Pods WITH CONFIGURATION Release ===
-
-    Check dependencies
-
-    blah
-
-    === CLEAN TARGET Pods-Runner OF PROJECT Pods WITH CONFIGURATION Release ===
-
-    Check dependencies
-
-    blah
-
-    === CLEAN TARGET Runner OF PROJECT Runner WITH CONFIGURATION Release ===
-
     Check dependencies
     [BCEROR]Signing for "Runner" requires a development team. Select a development team in the project editor.
-    [BCEROR]Code signing is required for product type 'Application' in SDK 'iOS 10.3'
-    [BCEROR]Code signing is required for product type 'Application' in SDK 'iOS 10.3'
-    [BCEROR]Code signing is required for product type 'Application' in SDK 'iOS 10.3'
-
-    blah
-
-    ** CLEAN SUCCEEDED **
-
-    === BUILD TARGET url_launcher OF PROJECT Pods WITH CONFIGURATION Release ===
-
-    Check dependencies
-
-    blah
-
-    === BUILD TARGET Pods-Runner OF PROJECT Pods WITH CONFIGURATION Release ===
-
-    Check dependencies
-
-    blah
-
-    === BUILD TARGET Runner OF PROJECT Runner WITH CONFIGURATION Release ===
-
-    Check dependencies
-    Signing for "Runner" requires a development team. Select a development team in the project editor.
-    Code signing is required for product type 'Application' in SDK 'iOS 10.3'
-    Code signing is required for product type 'Application' in SDK 'iOS 10.3'
-    Code signing is required for product type 'Application' in SDK 'iOS 10.3'
 
 Could not build the precompiled application for the device.''',
         xcodeBuildExecution: XcodeBuildExecution(
@@ -323,6 +283,47 @@ Could not build the precompiled application for the device.''',
         logger.errorText,
         contains('Building a deployable iOS app requires a selected Development Team with a \nProvisioning Profile.'),
       );
+    });
+
+    testWithoutContext('does not show no development team message when other Xcode issues detected', () async {
+      final XcodeBuildResult buildResult = XcodeBuildResult(
+        success: false,
+        stdout: '''
+Running "flutter pub get" in flutter_gallery...  0.6s
+Launching lib/main.dart on x in release mode...
+Running pod install...                                1.2s
+Running Xcode build...                                1.4s
+Failed to build iOS app
+Error output from Xcode build:
+↳
+    ** BUILD FAILED **
+
+
+    The following build commands failed:
+    	Check dependencies
+    (1 failure)
+Xcode's output:
+↳
+    blah
+
+    Check dependencies
+    [BCEROR]Signing for "Runner" requires a development team. Select a development team in the project editor.
+
+Could not build the precompiled application for the device.''',
+        xcodeBuildExecution: XcodeBuildExecution(
+          buildCommands: <String>['xcrun', 'xcodebuild', 'blah'],
+          appDirectory: '/blah/blah',
+          environmentType: EnvironmentType.physical,
+          buildSettings: buildSettings,
+        ),
+        xcResult: XCResult.test(issues: <XCResultIssue>[
+          XCResultIssue.test(message: 'Target aot_assembly_release failed', subType: 'Error'),
+        ])
+      );
+
+      await diagnoseXcodeBuildFailure(buildResult, testUsage, logger);
+      expect(logger.errorText, contains('Error (Xcode): Target aot_assembly_release failed'));
+      expect(logger.errorText, isNot(contains('Building a deployable iOS app requires a selected Development Team')));
     });
   });
 

--- a/packages/flutter_tools/test/integration.shard/flutter_build_with_compilation_error_test.dart
+++ b/packages/flutter_tools/test/integration.shard/flutter_build_with_compilation_error_test.dart
@@ -64,8 +64,7 @@ int x = 'String';
       ], workingDirectory: projectRoot.path);
 
       expect(
-        // iOS shows this as stdout.
-        targetPlatform == 'ios' ? result.stdout : result.stderr,
+        result.stderr,
         contains("A value of type 'String' can't be assigned to a variable of type 'int'."),
       );
       expect(result.exitCode, 1);

